extra/yoda

```diff
@@ -155,33 +168,86 @@ def parse_profile(arg, mintemp=0, maxtemp=100, minduty=0, maxduty=100):
 
     """
     try:
-        val = ast.literal_eval('[' + arg + ']')
-        if len(val) == 1 and isinstance(val[0], int):
-            # for arg == '<number>' set fixed duty between mintemp and maxtemp - 1
-            val = [(mintemp, val[0]), (maxtemp - 1, val[0])]
+        if arg in liquidctl.driver.msi.MpgCooler.BUILTIN_MODES.keys() and str_allowed:
+            return arg
+        else:
+            val = ast.literal_eval('[' + arg + ']')
+            if len(val) == 1 and isinstance(val[0], int):
+                # for arg == '<number>' set fixed duty between mintemp and maxtemp - 1
+                val = [(mintemp, val[0]), (maxtemp - 1, val[0])]
     except:
-        raise ValueError('profile must be comma-separated (temperature, duty) tuples')
+        msg = (
+            'profile must be comma-separated (temperature, duty) tuples or '
+            + f'recognised cooler mode string {str(liquidctl.driver.msi.MpgCooler.BUILTIN_MODES.keys())}'
+        )
+        raise ValueError(msg)
     for step in val:
         if not isinstance(step, tuple) or len(step) != 2:
             raise ValueError('profile must be comma-separated (temperature, duty) tuples')
         temp, duty = step
         if not isinstance(temp, int) or temp < mintemp or temp > maxtemp:
-            raise ValueError('temperature must be integer between {} and {}'.format(mintemp, maxtemp))
+            raise ValueError(
+                'temperature must be integer between {} and {}'.format(mintemp, maxtemp)
+            )
         if not isinstance(duty, int) or duty < minduty or duty > maxduty:
             raise ValueError('duty must be integer between {} and {}'.format(minduty, maxduty))
     return normalize_profile(val, critx=maxtemp)
 
 
+def auto_control(device, channels, profiles, sensors, update_interval):
```

Comment:
This is mostly geared towards the specific MSI AIO, leave a comment to note that.

Reply:
Thank you for the suggestion, added a note.